lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -279,6 +296,15 @@ InnerRefMap LowerLayersPass::runOnModuleLike(FModuleLike moduleLike) {
   return innerRefMap;
 }
 
+void LowerLayersPass::lowerInlineLayerBlock(LayerOp layer,
+                                            LayerBlockOp layerBlock) {
+  OpBuilder builder(layerBlock);
+  auto macroName = macroNames[layer];
+  auto ifDef = builder.create<sv::IfDefOp>(layerBlock.getLoc(), macroName);
+  ifDef.getBodyRegion().takeBody(layerBlock.getBodyRegion());
+  layerBlock.erase();
```

Comment:
I believe this is safe _even if it doesn't advance_ because this is walking post-order. See: https://mlir.llvm.org/doxygen/classmlir_1_1Operation.html#a59740592240b950b8c8afcf4a2eb4113

Reply:
Sure, I'm familiar and yep you're right (and I did miss that detail, whoops!).  Anyway the point is this is fine even if it's structured such that the legality isn't locally evident.